### PR TITLE
FPS Limiting - UI Redesign

### PR DESCRIPTION
**Pull request by adityaruplaha**
Follow up of #3523 .
Will do the actual hotkey work (#3417) in another PR.
Thanks to @jroweboy and @B3n30 for the UI design.

# Design changes:
Removed speed limit increase/decrease hotkeys.
Made 3 new functions:
 - A function to convert the setting from `QString` to a `u16`.
 - A function to do the reverse.
 - A function to automatically add the edited custom %age to the `QComboBox` items list. Users can enter these manually, if they don't exist already. If they exist, switch to that. (Tested - SUCCESS)

# Glitches:

Whenever "Unlimited" is selected, it glitches out into a random number in the UI, upon closing the config. Gameplay is unaffected, i.e. works as intended. No idea what may be causing it.

# To Do:
- [ ] Save those custom limits somewhere. (in `UISettings` maybe?)
For now, they appear just below "Unlimited" in the list for the current session. If Citra exits while a custom value is selected, then the custom value is saved in the INI. Upon reloading, this value is loaded & appears at the end of the item list. All other custom values are deleted. (Tested - SUCCESS)
(See the second screenshot)

- [x] Make "Unlimited" non case sensitive. (Tested - SUCCESS)

# Screenshots
image

image

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3532)

### CODE REVIEW

**Comment by adityaruplaha** [comment]
I'm effectively done. Except that little glitch with "Unlimited".

**Comment by adityaruplaha** [comment]
@chris062689 I've modified telemetry. Please check it out.

**Comment by chris062689** [comment]
Were any configuration items added? Those will need to be added to telemetry. I only see fields removed.

**Comment by adityaruplaha** [comment]
Nope, nothing added. Only removed the boolean.

**Comment by adityaruplaha** [comment]
I don't know anything about data roles, but sure, I'll do some research in the Qt Docs & find out if I can use them. Please add `pr:needs-work`.

File: src/core/perf_stats.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ double PerfStats::GetLastFrameTimeScale() {
 }
 
 void FrameLimiter::DoFrameLimiting(u64 current_system_time_us) {
-    if (!Settings::values.use_frame_limit) {
+    if (Settings::values.frame_limit == Settings::NO_FRAME_LIMITING) {
```
Comment (B3n30) [inline src/core/perf_stats.cpp]: I don't think thats the behavior I and @jroweboy had in mind. And it's probably also not the behavior you had in mind. The idea is: if `use_frame_limit ` is false have a frame_limit from 100%. if it is true use the `Settings::values.frame_limit` as a frame limit. And only if use_frame_limit is true **and** frame_limit is set to NO_FRAME_LIMITING return at this point.

If you do that the game will always run at 100%, but if the frame_limit hotkey gets pressed you get a speed_up/slow_down by whatever the setting is set to.

File: src/citra_qt/configuration/configure_graphics.ui [hunk 2]
```diff
@@ -46,26 +46,115 @@
...
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>75%</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>100%</string>
```
Comment (B3n30) [inline src/citra_qt/configuration/configure_graphics.ui]: I think if you apply the changes I stated in the other comment 100% isn't necessary here

File: src/citra_qt/configuration/configure_graphics.ui [hunk 3]
```diff
@@ -46,26 +46,115 @@
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
-           <widget class="QCheckBox" name="toggle_frame_limit">
+           <widget class="QLabel" name="label_2">
             <property name="text">
-             <string>Limit Speed Percent</string>
+             <string>FPS Limit:</string>
```
Comment (B3n30) [inline src/citra_qt/configuration/configure_graphics.ui]: I don't like the name FPS Limit. Maybe something like `speed hotkey limit`?

**Comment by adityaruplaha** [comment]
Well, I'll discuss this in Discord. I think I may have got what you are trying to say.

**Comment by adityaruplaha** [comment]
I'll rewrite the entire code (yet again). Closing

**Pull request closed without merging**